frontend/src/metabase/lib/expressions/process.js

```diff
@@ -1,31 +1,45 @@
-// combine compile/suggest/syntax so we only need to parse once
+import { parse } from "metabase/lib/expressions/recursive-parser";
+import { resolve } from "metabase/lib/expressions/resolver";
+
+import {
+  parseDimension,
+  parseMetric,
+  parseSegment,
+} from "metabase/lib/expressions/";
+
 export function processSource(options) {
-  // Lazily load all these parser-related stuff, because parser construction is expensive
-  // https://github.com/metabase/metabase/issues/13472
-  const parse = require("./parser").parse;
-  const compile = require("./compile").compile;
+  const resolveMBQLField = (kind, name) => {
```

Comment:
Should we break this out into an exported function so that the pratt parser can share it?

Reply:
Good point, though I think this `process.js` might disappear anyway now that the editor itself has been refactored and looks like we don't need a dedicated `processSource` function.